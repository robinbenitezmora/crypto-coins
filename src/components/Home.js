import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchCryptoCurrencies } from '../redux/cryptoreducer';
import CryptoElementList from './CryptoList';

const Home = () => {
  const [search, setSearch] = useState('');
  const dispatch = useDispatch();
  const cryptos = useSelector((state) => state.crypto.cryptosList);

  useEffect(() => {
    dispatch(fetchCryptoCurrencies());
  }, []);

  useEffect(() => {
    dispatch(fetchCryptoCurrencies());
  }, []);

  const handleChange = (e) => {
    setSearch(e.target.value);
  };

  return (
    <div className="home">
      <div className="homeSearch">
        <input type="text" placeholder="Search" onChange={handleChange} value={search} />
      </div>
      <CryptoElementList cryptos={cryptos} search={search} />
    </div>
  );
};

export default Home;
