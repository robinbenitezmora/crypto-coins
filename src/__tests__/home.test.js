import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Home from '../components/Home';

test('Home renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
