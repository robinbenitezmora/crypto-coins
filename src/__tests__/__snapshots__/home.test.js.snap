// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="home"
>
  <div
    className="homeSearch"
  >
    <input
      onChange={[Function]}
      placeholder="Search your Crypto"
      type="text"
      value=""
    />
  </div>
  <div
    className="cryptoList"
  >
    <h3
      className="cryptoListTitle"
    >
      CRYPTO COINS CURRENT RANKING
    </h3>
    <ul
      className="cryptoListContainer"
    />
  </div>
</div>
`;
